tests/promptness.rs

```diff
@@ -0,0 +1,205 @@
+extern crate tiny_http;
+
+use std::io::{copy, Read, Write};
+use std::net::{Shutdown, TcpStream};
+use std::ops::Deref;
+use std::sync::mpsc::channel;
+use std::sync::Arc;
+use std::thread::{sleep, spawn};
+use std::time::Duration;
+use tiny_http::{Response, Server};
+
+/// Stream that produces bytes very slowly
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct SlowByteSrc {
+    val: u8,
+    len: usize,
+}
+impl<'b> Read for SlowByteSrc {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        sleep(Duration::from_millis(100));
+        let l = self.len.min(buf.len()).min(1000);
+        buf[..l].fill(self.val);
```

Comment:
I appreciate this is just inside test code but it breaks under 1.41.1 because `.fill()` wasn't yet implemented for `[u8]`.

https://github.com/tiny-http/tiny-http/pull/207/checks?check_run_id=2940126934